packages/main/src/RPA/MSGraph.py

```diff
@@ -0,0 +1,956 @@
+import base64
+import logging
+import importlib
+from typing import Dict, Optional, Union
+from pathlib import Path
+from O365 import (
+    Account,
+    MSGraphProtocol,
+    FileSystemTokenBackend,
+    directory,
+    drive,
+    sharepoint,
+)
+from O365.utils import Token, BaseTokenBackend
+from O365.utils.utils import (  # noqa: F401 pylint: disable=unused-import
+    ME_RESOURCE,
+    USERS_RESOURCE,
+    GROUPS_RESOURCE,
+    SITES_RESOURCE,
+)
+from robot.api.deco import keyword
+
+
+DEFAULT_REDIRECT_URI = "https://login.microsoftonline.com/common/oauth2/nativeclient"
+DEFAULT_TOKEN_PATH = Path("/temp")
+DEFAULT_PROTOCOL = MSGraphProtocol()
+
+
+def import_table():
+    """Try to import `RPA.Tables.Table`"""
+    try:
+        module = importlib.import_module("RPA.Tables")
+        return getattr(module, "Table")
+    except ModuleNotFoundError:
+        return None
+
+
+Table = import_table()
+DataTable = Table if Table else list[Dict]
+
+
+class MSGraphAuthenticationError(Exception):
+    "Error when authentication fails."
+
+
+class MSGraphDownloadError(Exception):
+    "Error when download fails."
+
+
+class RobocorpVaultTokenBackend(BaseTokenBackend):
+    "A simple Token backend that saves to Robocorp vault"
+
+
+class SharedItem(drive.File):
+    """Simple class to add support for shared items.
+    Inherits File only to bypass checks in the library.
+    """  # pylint: disable=super-init-not-called
+
+    def __init__(self, data, url, con):
+        self.name = data["name"]
+        self.size = data["size"]
+        self.con = con
+        self.url = url
+        self.object_id = None
+
+    def build_url(self, *args):  # pylint: disable=unused-argument
+        """Discard arguments since we already have the correct URL."""
+        return self.url
+
+
+class MSGraph:
```

Comment:
in overall in keyword descriptions some define `:return:` and some don't, better to have consistent style within one library

Reply:
Added to all that I saw missing.